active/0000-data-loading-methods.md

```diff
@@ -0,0 +1,68 @@
+- Start Date: 12/22/2014
+- RFC PR: 
+- Ember Issue: 
+
+# Summary
+
+I believe that the data loading methods in Ember Data should be broken down and rearranged based on the methods by which they load and return data, whether it is from a local store, from a remote server, or a combination of both. The names of each of these methods should also be carefully considered (likely using the same name for both single object and collection loads) so as to avoid confusion and reduce any learning curves.
+
+# Motivation
+
+Currently in Ember Data the layout and naming of the data loading functions is somewhat disorganized, which increases the mental overhead of using the library. `.all` and `.getById` return local data, `.find` may look locally or remote, except for when you pass an object (i.e. `.find({minAge: '1'})`) in which case the server is always queried. There's also a new `.fetch` function that will always request new data from the store, and possibly other methods that just aren't coming to mind at the moment.
+
+# Detailed design
+
+The current scheme for data loading in Ember Data is something like:
+
+|multiplicity|local data|fetch if not local|fetch always|local, with new load in background|
+|---|---|---|---|---|
+|collection|store.all|?|store.find (query)|?|
```

Comment:
shouldn't it be something like this?

|multiplicity|local data|fetch if not local|fetch always|local, with new load in background|
|---|---|---|---|---|
|collection|store.all|store.find|store.find (query)|?|
|single record|store.getById|store.find|store.fetch|?|